packages/engine/lib/orchestrator/src/experiment.rs

```diff
@@ -276,10 +276,21 @@ impl Experiment {
             env: ExecutionEnvironment::None, // We don't connect to the API
             dyn_payloads: serde_json::Map::from_iter(map_iter),
         };
-        engine_process
+        if let Err(err) = engine_process
             .send(&proto::EngineMsg::Init(init_message))
             .await
-            .wrap_err("Could not send `Init` message")?;
+            .wrap_err("Could not send `Init` message")
+        {
+            // TODO: Wait for threads to finish before starting a forced cleanup
+            warn!("Engine didn't exit gracefully, waiting for threads to finish.");
```

Comment:
Threads shouldn't run anymore at this point. Python is started in a subprocess, so this is why we need this sleep.
```suggestion
            warn!("Engine didn't exit gracefully, waiting for subprocesses to finish.");
```